File: VTI_Glaxy_FE/src/services/bookingService.js
import axiosClient from "./axiosClient";

const bookingService = {
  fetchBookings() {
    return axiosClient.get("/getBookings");
  },
  fetchBookingById(bookingId) {
    return axiosClient.get(`/getBookingById?bookingId=${bookingId}`);
  },
  createBooking(bookingRequest) {
    return axiosClient.post("/postBooking", bookingRequest);
  },
  updateBooking(bookingId, bookingRequest) {
    return axiosClient.put(
      `/putBooking?bookingId=${bookingId}`,
      bookingRequest
    );
  },
  deleteBooking(bookingId) {
    return axiosClient.put(`/deleteBooking?bookingId=${bookingId}`);
  },
};

export default bookingService;

File: VTI_Galaxy_BE/src/main/java/demo/services/interfaceClass/BookingService.java
package demo.services.interfaceClass;

import demo.modal.dto.BookingDto;
import demo.modal.request.BookingRequest;

import java.util.List;

public interface BookingService {
    List<BookingDto> getBookings();

    BookingDto getBookingById(int id);

    BookingDto createBooking(BookingRequest request);

    BookingDto updateBooking(int id, BookingRequest request);

    BookingDto deleteBooking(int id);

    List<BookingDto> getBookingsByAccountId(int accountId);
}
